src/spikeinterface/postprocessing/template_metrics.py

```diff
@@ -141,8 +145,29 @@ def _set_params(
             metrics_kwargs_ = _default_function_kwargs.copy()
             metrics_kwargs_.update(metrics_kwargs)
 
+        all_metric_names = metric_names
+        tm_extension = self.sorting_analyzer.get_extension("template_metrics")
+        if delete_existing_metrics is False and tm_extension is not None:
+            existing_metric_names = tm_extension.params["metric_names"]
+            existing_params = tm_extension.params["metrics_kwargs"]
+
+            # checks that existing metrics were calculated using the same params
+            if existing_params != metrics_kwargs_:
+                warnings.warn(
+                    "The parameters used to calculate the previous template metrics are different than those used now. Deleting previous template metrics..."
+                )
+                self.sorting_analyzer.get_extension("template_metrics").data["metrics"] = pd.DataFrame(
+                    index=self.sorting_analyzer.unit_ids
+                )
+                self.sorting_analyzer.get_extension("template_metrics").params["metric_names"] = []
+                existing_metric_names = []
+
+            for metric_name in existing_metric_names:
```

Comment:
Is my understanding correct that the existing metrics are re-computed? they are added to the params dict here, and then the call `_run` -> `_compute_metrics` will re-compute everything? (or, maybe there is some skipping of existing templates in the `compute_metrics` logic I have missed)

Reply:
They were but that was a mistake. Hopefully clearer now.